src/UI/state/map/mapSlice.ts

```diff
@@ -0,0 +1,404 @@
+  "okuensis",
+  "berghei",
+  "mortiauxi",
+  "funestus-like",
+  "arabiensis",
+  "culicifacies",
+  "fuscivenosus",
+  "quadriannulatus",
+  "dthali",
+  "domicola",
+  "fontinalis",
+  "squamosus",
+  "hancocki",
+  "faini",
+  "maculipalpis",
+  "funestus",
+  "buxtoni",
+  "dureni",
+  "jebudensis",
+  "rhodesiensis",
+  "grassei",
+  "paludis",
+  "seretsei",
+  "erythraeus",
+  "eouzani",
+  "notleyi",
+  "gabonensis",
+  "hargreavesi",
+  "dancalicus",
+  "somalicus",
+  "longipalpis",
+  "kingi",
+  "vernus",
+  "cameroni",
+  "fontenillei",
+  "hervyi",
+  "brucei",
+  "ranci",
+  "vinckei",
+  "rivulorum-like",
+  "radama",
+  "erepens",
+  "confusus",
+  "austenii",
+  "letabensis",
+  "cydippis",
+  "seydeli",
+  "deemingi",
+  "carteri",
+  "wellcomei",
+  "christyi",
+  "concolor",
+  "keniensis",
+  "ruarinus",
+  "moucheti",
+  "grenieri",
+  "mascarensis",
+  "barberellus",
+  "stephensi",
+  "gibbinsi",
+  "lovettae",
+  "flavicosta",
+  "murphyi",
+  "milloti",
+  "azevedoi",
+  "turkhudi",
+  "griveaudi",
+  "brohieri",
+  "roubaudi",
+  "mousinhoi",
+  "daudi",
+  "multicolor",
+  "multicinctus",
+  "caroni",
+  "carnevalei",
+  "swahilicus",
+  "cinereus",
+  "harperi",
+  "aruni",
+  "fuscicolor",
+  "ardensis",
+  "nili",
+  "rufipes",
+  "freetownensis",
+  "machardyi",
+  "obscurus",
+  "hamoni",
+  "ziemanni",
+  "dualaensis",
+  "vaneedeni",
+  "rivulorum",
+  "walravensi",
+  "garnhami",
+  "vanhoofi",
+  "kosiensis",
+  "demeilloni",
+  "argenteolobatus",
+  "schwetzi",
+  "natalensis",
+  "marshallii",
+  "ethiopicus",
+  "lacani",
+  "bwambae",
+  "pharoensis",
+  "namibiensis",
+  "implexus",
+  "parensis",
+  "merus",
+  "distinctus",
+  "pretoriensis",
+  "hughi",
+  "gambiae",
+  "symesi",
+  "listeri",
+  "lounibosi",
+  "caliginosus",
+  "lloreti",
+  "tenebrosus",
+  "smithii",
+  "wilsoni",
+  "maliensis",
+  "azaniae",
+  "salbaii",
+  "crypticus",
+  "sergentii",
+  "theileri",
+  "amharicus",
+  "millecampsi",
+  "rageaui",
+  "brunnipes",
+  "brumpti",
+  "leesoni",
+  "tchekedii",
+  "melas",
+  "coluzzii",
+  "cristipalpis",
+  "rodhaini",
+];
+
+function singularOutputs(filters: VectorAtlasFilters) {
+  const updatedFilters = JSON.parse(JSON.stringify(filters));
+  updatedFilters.country.value = filters.country.value[0];
+  updatedFilters.species.value = filters.species.value[0];
+  updatedFilters.season.value = filters.season.value[0];
+  updatedFilters.control.value = filters.control.value[0];
+  updatedFilters.isAdult.value = filters.isAdult.value[0];
+  updatedFilters.isLarval.value = filters.isLarval.value[0];
+  return updatedFilters;
+}
+
+export interface MapState {
+  map_styles: {
+    layers: {
+      name: string;
+      fillColor: number[];
+      strokeColor: number[];
+      strokeWidth: number;
+      zIndex: number;
+    }[];
+  };
+  map_overlays: {
+    name: string;
+    sourceLayer: string;
+    sourceType: string;
+    isVisible: boolean;
+  }[];
+  currentSearchID: string;
```

Comment:
It means that if there is a paged query already running then the UI ignores any further results coming back from the already running queries. Otherwise you end up with chaos from the race conditions of 2 sets of queries running at the same time. The alternative would be to work out how to include cancellable promises instead.